# ML Mini Project 

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
# Import data set

data = pd.read_csv("tmdb-movies (1).csv")   
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
id                      10866 non-null int64
imdb_id                 10856 non-null object
popularity              10866 non-null float64
budget                  10866 non-null int64
revenue                 10866 non-null int64
original_title          10866 non-null object
cast                    10790 non-null object
homepage                2936 non-null object
director                10822 non-null object
tagline                 8042 non-null object
keywords                9373 non-null object
overview                10862 non-null object
runtime                 10866 non-null int64
genres                  10843 non-null object
production_companies    9836 non-null object
release_date            10866 non-null object
vote_count              10866 non-null int64
vote_average            10866 non-null float64
release_year            10866 non-null int64
budget_adj              1

In [4]:
data.head(5)

id    imdb_id  popularity     budget     revenue  \
0  135397  tt0369610   32.985763  150000000  1513528810   
1   76341  tt1392190   28.419936  150000000   378436354   
2  262500  tt2908446   13.112507  110000000   295238201   
3  140607  tt2488496   11.173104  200000000  2068178225   
4  168259  tt2820852    9.335014  190000000  1506249360   

                 original_title  \
0                Jurassic World   
1            Mad Max: Fury Road   
2                     Insurgent   
3  Star Wars: The Force Awakens   
4                     Furious 7   

                                                cast  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   

                                            homepage          director  \
0                      http://www.jurassicworld.com/   Colin Trevorrow   
1                        http://www.madmaxmovie.com/     George Miller   
2     http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   
3  http://www.starwars.com/films/star-wars-episod...       J.J. Abrams   
4                           http://www.furious7.com/         James Wan   

                         tagline      ...       \
0              The park is open.      ...        
1             What a Lovely Day.      ...        
2     One Choice Can Destroy You      ...        
3  Every generation has a story.      ...        
4            Vengeance Hits Home      ...        

                                            overview runtime  \
0  Twenty-two years after the events of Jurassic ...     124   
1  An apocalyptic story set in the furthest reach...     120   
2  Beatrice Prior must confront her inner demons ...     119   
3  Thirty years after defeating the Galactic Empi...     136   
4  Deckard Shaw seeks revenge against Dominic Tor...     137   

                                      genres  \
0  Action|Adventure|Science Fiction|Thriller   
1  Action|Adventure|Science Fiction|Thriller   
2         Adventure|Science Fiction|Thriller   
3   Action|Adventure|Science Fiction|Fantasy   
4                      Action|Crime|Thriller   

                                production_companies release_date vote_count  \
0  Universal Studios|Amblin Entertainment|Legenda...       6/9/15       5562   
1  Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15       6185   
2  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15       2480   
3          Lucasfilm|Truenorth Productions|Bad Robot     12/15/15       5292   
4  Universal Pictures|Original Film|Media Rights ...       4/1/15       2947   

   vote_average  release_year    budget_adj   revenue_adj  
0           6.5          2015  1.379999e+08  1.392446e+09  
1           7.1          2015  1.379999e+08  3.481613e+08  
2           6.3          2015  1.012000e+08  2.716190e+08  
3           7.5          2015  1.839999e+08  1.902723e+09  
4           7.3          2015  1.747999e+08  1.385749e+09  

[5 rows x 21 columns]

In [5]:
data.describe()

id    popularity        budget       revenue       runtime  \
count   10866.000000  10866.000000  1.086600e+04  1.086600e+04  10866.000000   
mean    66064.177434      0.646441  1.462570e+07  3.982332e+07    102.070863   
std     92130.136561      1.000185  3.091321e+07  1.170035e+08     31.381405   
min         5.000000      0.000065  0.000000e+00  0.000000e+00      0.000000   
25%     10596.250000      0.207583  0.000000e+00  0.000000e+00     90.000000   
50%     20669.000000      0.383856  0.000000e+00  0.000000e+00     99.000000   
75%     75610.000000      0.713817  1.500000e+07  2.400000e+07    111.000000   
max    417859.000000     32.985763  4.250000e+08  2.781506e+09    900.000000   

         vote_count  vote_average  release_year    budget_adj   revenue_adj  
count  10866.000000  10866.000000  10866.000000  1.086600e+04  1.086600e+04  
mean     217.389748      5.974922   2001.322658  1.755104e+07  5.136436e+07  
std      575.619058      0.935142     12.812941  3.430616e+07  1.446325e+08  
min       10.000000      1.500000   1960.000000  0.000000e+00  0.000000e+00  
25%       17.000000      5.400000   1995.000000  0.000000e+00  0.000000e+00  
50%       38.000000      6.000000   2006.000000  0.000000e+00  0.000000e+00  
75%      145.750000      6.600000   2011.000000  2.085325e+07  3.369710e+07  
max     9767.000000      9.200000   2015.000000  4.250000e+08  2.827124e+09

In [5]:
data.isna().sum()

id                         0
imdb_id                   10
popularity                 0
budget                     0
revenue                    0
original_title             0
cast                      76
homepage                7930
director                  44
tagline                 2824
keywords                1493
overview                   4
runtime                    0
genres                    23
production_companies    1030
release_date               0
vote_count                 0
vote_average               0
release_year               0
budget_adj                 0
revenue_adj                0
dtype: int64

In [7]:
# Fill Null values with 0

data.fillna(0)

id    imdb_id  popularity     budget     revenue  \
0      135397  tt0369610   32.985763  150000000  1513528810   
1       76341  tt1392190   28.419936  150000000   378436354   
2      262500  tt2908446   13.112507  110000000   295238201   
3      140607  tt2488496   11.173104  200000000  2068178225   
4      168259  tt2820852    9.335014  190000000  1506249360   
5      281957  tt1663202    9.110700  135000000   532950503   
6       87101  tt1340138    8.654359  155000000   440603537   
7      286217  tt3659388    7.667400  108000000   595380321   
8      211672  tt2293640    7.404165   74000000  1156730962   
9      150540  tt2096673    6.326804  175000000   853708609   
10     206647  tt2379713    6.200282  245000000   880674609   
11      76757  tt1617661    6.189369  176000003   183987723   
12     264660  tt0470752    6.118847   15000000    36869414   
13     257344  tt2120120    5.984995   88000000   243637091   
14      99861  tt2395427    5.944927  280000000  1405035767   
15     273248  tt3460252    5.898400   44000000   155760117   
16     260346  tt2446042    5.749758   48000000   325771424   
17     102899  tt0478970    5.573184  130000000   518602163   
18     150689  tt1661199    5.556818   95000000   542351353   
19     131634  tt1951266    5.476958  160000000   650523427   
20     158852  tt1964418    5.462138  190000000   209035668   
21     307081  tt1798684    5.337064   30000000    91709827   
22     254128  tt2126355    4.907832  110000000   470490832   
23     216015  tt2322441    4.710402   40000000   569651467   
24     318846  tt1596363    4.648046   28000000   133346506   
25     177677  tt2381249    4.566713  150000000   682330139   
26     214756  tt2637276    4.564549   68000000   215863606   
27     207703  tt2802144    4.503789   81000000   403802136   
28     314365  tt1895587    4.062293   20000000    88346473   
29     294254  tt4046784    3.968891   61000000   311256926   
...       ...        ...         ...        ...         ...   
10836   38720  tt0061170    0.239435          0           0   
10837   19728  tt0060177    0.291704          0           0   
10838   22383  tt0060862    0.151845          0           0   
10839   13353  tt0060550    0.276133          0           0   
10840   34388  tt0060437    0.102530          0           0   
10841   42701  tt0062262    0.264925      75000           0   
10842   36540  tt0061199    0.253437          0           0   
10843   29710  tt0060588    0.252399          0           0   
10844   23728  tt0059557    0.236098          0           0   
10845    5065  tt0059014    0.230873          0           0   
10846   17102  tt0059127    0.212716          0           0   
10847   28763  tt0060548    0.034555          0           0   
10848    2161  tt0060397    0.207257    5115000    12000000   
10849   28270  tt0060445    0.206537          0           0   
10850   26268  tt0060490    0.202473          0           0   
10851   15347  tt0060182    0.342791          0           0   
10852   37301  tt0060165    0.227220          0           0   
10853   15598  tt0060086    0.163592          0           0   
10854   31602  tt0060232    0.146402          0           0   
10855   13343  tt0059221    0.141026     700000           0   
10856   20277  tt0061135    0.140934          0           0   
10857    5921  tt0060748    0.131378          0           0   
10858   31918  tt0060921    0.317824          0           0   
10859   20620  tt0060955    0.089072          0           0   
10860    5060  tt0060214    0.087034          0           0   
10861      21  tt0060371    0.080598          0           0   
10862   20379  tt0060472    0.065543          0           0   
10863   39768  tt0060161    0.065141          0           0   
10864   21449  tt0061177    0.064317          0           0   
10865   22293  tt0060666    0.035919      19000           0   

                                         original_title  \
0                                        Jura

## Data Cleaning


## Removing duplicate rows

In [10]:
data.drop_duplicates(inplace = True)
print("Afetr Removing Duplicate Values (Rows,Columns) : ",data.shape)

Afetr Removing Duplicate Values (Rows,Columns) :  (10865, 21)


##  Changing Format Of Release Date to Datetime Format

In [11]:
data['release_date'] = pd.to_datetime(data['release_date'])
data['release_date'].head()

0   2015-06-09
1   2015-05-13
2   2015-03-18
3   2015-12-15
4   2015-04-01
Name: release_date, dtype: datetime64[ns]

#  Removing columns that are not needed in the analysis process

In [12]:
data.drop(['budget_adj','revenue_adj','overview','imdb_id','homepage','tagline'],axis =1,inplace = True)
print("Afetr Removing Unused Columns (Rows,Columns) : ",data.shape)

Afetr Removing Unused Columns (Rows,Columns) :  (10865, 15)


## Droping rows containing 0 value

In [11]:
print("Rows With Zero Values In The Budget Column:",data[(data['budget']==0)].shape[0])
print("Rows With Zero Values In The Revenue Column:",data[(data['revenue']==0)].shape[0])

Rows With Zero Values In The Budget Column: 5696
Rows With Zero Values In The Revenue Column: 6016


# Perform EDA(Exploratory Data Analysis)

##  Q1. Which are the movies with the third lowest and third highest budget?

In [15]:
df1=data.drop(columns=['id', 'popularity','revenue','cast', 'director', 'keywords','runtime', 'genres', 'production_companies', 'release_date','vote_count', 'vote_average', 'release_year'])
df1=df1.sort_values('budget')
df1

budget                                original_title
4353          0                    Hercules in the Underworld
5149          0                                      Chouchou
9711          0                             The Secret Garden
5146          0                                  Hope Springs
5144          0                      Hitler: The Rise of Evil
5143          0                             American Splendor
5141          0                                Harvie Krumpet
5140          0                    The Ghost of Lord Farquaad
5139          0                                The Good Thief
9712          0                                Hope and Glory
5137          0                             De Zaak Alzheimer
5150          0                                        Fear X
5136          0                                 The Big Empty
5134          0                       Capturing the Friedmans
5131          0                            My Life Without Me
5129          0                                      Dead End
5127          0                                 The Statement
5126          0     Scooby-Doo! And the Legend of the Vampire
5125          0                                 The Challenge
9715          0                     My Best Friend's Birthday
5120          0                                       Absolon
5119          0         Scooby-Doo! and the Monster of Mexico
5118          0                       Les Enfants de la Pluie
9713          0                                 The Principal
5117          0                               Mimic: Sentinel
5151          0     Aileen: Life and Death of a Serial Killer
5153          0                          After School Special
5185          0             Dickie Roberts: Former Child Star
9708          0                              Miami Connection
5183          0                           Bright Young Things
...         ...                                           ...
654   200000000                      The Amazing Spider-Man 2
3394  200000000                                 Green Lantern
5444  200000000                           Monsters University
6215  207000000                                     King Kong
4405  209000000                                    Battleship
648   210000000               Transformers: Age of Extinction
6883  210000000                         X-Men: The Last Stand
4370  215000000                        The Amazing Spider-Man
4361  220000000                                  The Avengers
2902  225000000      The Chronicles of Narnia: Prince Caspian
5432  225000000                                  Man of Steel
4381  225000000                                Men in Black 3
1386  237000000                                        Avatar
10    245000000                                       Spectre
643   250000000                    X-Men: Days of Future Past
1389  250000000        Harry Potter and the Half-Blood Prince
5431  250000000           The Hobbit: The Desolation of Smaug
634   250000000     The Hobbit: The Battle of the Five Armies
1923  250000000  Harry Potter and the Deathly Hallows: Part 1
4367  250000000             The Hobbit: An Unexpected Journey
4363  250000000                         The Dark Knight Rises
5508  255000000                               The Lone Ranger
7394  258000000                                  Spider-Man 3
1929  260000000                                       Tangled
4411  260000000                                   John Carter
6570  270000000                              Superman Returns
14    280000000                       Avengers: Age of Ultron
7387  300000000      Pirates of the Caribbean: At World's End
3375  380000000   Pirates of the Caribbean: On Stranger Tides
2244  425000000                             The Warrior's Way

[10865 rows x 2 columns]

In [16]:
# Removing movies with 0 budget

df1=df1[(df1['budget']>0)]
df1

budget                                original_title
3065           1            Angus, Thongs and Perfect Snogging
2618           1                                  Lost & Found
1151           1                                   Fear Clinic
3581           1                       Love, Wedding, Marriage
8944           2                                  Death Wish 2
10050          3            Tales from the Darkside: The Movie
2398           3                                           Boy
3765           3                          Death of a Superhero
1313           5                   Des Lendemains Qui Chantent
8235           6                           Tales from the Hood
3015           8                                     Max Manus
8125           8                                  The Prophecy
10297          8            National Lampoon's Loaded Weapon 1
981           10                                       Plastic
6460          10                                        Edmond
586           10                                     Penis Boy
2381          10                   Once Upon a Time in Mumbaai
5162          10                                   Kid's Story
5852          10                                  Hross í oss
10777         11                                      F.I.S.T.
10316         12                             The Tommyknockers
7694          12                                   Delta Farce
7195          14                                    Modigliani
5785          15                                Make Your Move
5956          15                                       Peeples
4705          17                Ce que le jour doit à la nuit
5949          18                          Prisoners of the Sun
1141          18                             Enchanted Kingdom
3853          20                                William & Kate
10236         21              Teenage Mutant Ninja Turtles III
...          ...                                           ...
654    200000000                      The Amazing Spider-Man 2
3394   200000000                                 Green Lantern
5444   200000000                           Monsters University
6215   207000000                                     King Kong
4405   209000000                                    Battleship
648    210000000               Transformers: Age of Extinction
6883   210000000                         X-Men: The Last Stand
4370   215000000                        The Amazing Spider-Man
4361   220000000                                  The Avengers
2902   225000000      The Chronicles of Narnia: Prince Caspian
5432   225000000                                  Man of Steel
4381   225000000                                Men in Black 3
1386   237000000                                        Avatar
10     245000000                                       Spectre
643    250000000                    X-Men: Days of Future Past
1389   250000000        Harry Potter and the Half-Blood Prince
5431   250000000           The Hobbit: The Desolation of Smaug
634    250000000     The Hobbit: The Battle of the Five Armies
1923   250000000  Harry Potter and the Deathly Hallows: Part 1
4367   250000000             The Hobbit: An Unexpected Journey
4363   250000000                         The Dark Knight Rises
5508   255000000                               The Lone Ranger
7394   258000000                                  Spider-Man 3
1929   260000000                                       Tangled
4411   260000000                                   John Carter
6570   270000000                              Superman Returns
14     280000000                       Avengers: Age of Ultron
7387   300000000      Pirates of the Caribbean: At World's End
3375   380000000   Pirates of the Caribbean: On Stranger Tides
2244   425000000                             The Warrior's Way

[5169 rows x 2 columns]

In [17]:
df1.head(10)

budget                      original_title
3065        1  Angus, Thongs and Perfect Snogging
2618        1                        Lost & Found
1151        1                         Fear Clinic
3581        1             Love, Wedding, Marriage
8944        2                        Death Wish 2
10050       3  Tales from the Darkside: The Movie
2398        3                                 Boy
3765        3                Death of a Superhero
1313        5         Des Lendemains Qui Chantent
8235        6                 Tales from the Hood

## Ans. There are three movies named, "Death of a Superhero" , "Boy" , "Tales from the Darkside: The Movie" with third lowest budget worth 3 . The movie with third highest budget as 300000000 is "Pirates of the Caribbean : At World's End".

 ## Q4. Which are the movies with most and least earned revenue?

In [36]:
def find_minmax(x):    
    #use the function 'idxmin' to find the index of lowest profit movie.    
    min_index = data[x].idxmin()
    
    #use the function 'idxmax' to find the index of Highest profit movie.
    high_index = data[x].idxmax()
    
    high = pd.DataFrame(data.loc[high_index,:])
    low = pd.DataFrame(data.loc[min_index,:])
    print("Movie with Highest "+ x + " : ",data['original_title'][high_index])
    print("Movie with Lowest "+ x + "  : ",data['original_title'][min_index])
    
    return pd.concat([high,low],axis = 1)

find_minmax('revenue')

Movie with Highest revenue :  Avatar
Movie with Lowest revenue  :  Wild Card


1386  \
id                                                                19995   
popularity                                                      9.43277   
budget                                                        237000000   
revenue                                                      2781505847   
original_title                                                   Avatar   
cast                  Sam Worthington|Zoe Saldana|Sigourney Weaver|S...   
director                                                  James Cameron   
keywords              culture clash|future|space war|space colony|so...   
runtime                                                             162   
genres                         Action|Adventure|Fantasy|Science Fiction   
production_companies  Ingenious Film Partners|Twentieth Century Fox ...   
release_date                                        2009-12-10 00:00:00   
vote_count                                                         8458   
vote_average                                                        7.1   
release_year                                                       2009   

                                                                   48    
id                                                               265208  
popularity                                                      2.93234  
budget                                                         30000000  
revenue                                                               0  
original_title                                                Wild Card  
cast                  Jason Statham|Michael Angarano|Milo Ventimigli...  
director                                                     Simon West  
keywords                                      gambling|bodyguard|remake  
runtime                                                              92  
genres                                             Thriller|Crime|Drama  
production_companies  Current Entertainment|Lionsgate|Sierra / Affin...  
release_date                                        2015-01-14 00:00:00  
vote_count                                                          481  
vote_average                                                        5.3  
release_year                                                       2015

## Ans. The movie with Highest revenue is Avatar and  the movie with lowest revenue  is Wild Card


## Q5.  What is the average runtime of movies in the year 2006?

In [19]:
# Replacing 0 value by NAN

data['runtime'] = data['runtime'].replace(0,np.NAN)

In [20]:
#Removing not so important columns

df5=data.drop(columns=['id', 'popularity','cast', 'director','keywords','genres', 'production_companies', 'release_date','vote_count', 'vote_average'])
df5=df5[(df5['release_year']==2006)]
df5

budget     revenue  \
6554   50000000   111340801   
6555  200000000  1065659812   
6556  120000000   461983149   
6557  150000000   599045960   
6558  125000000   758239851   
6559   12000000   114194847   
6560   80000000   660940780   
6561  150000000   397850012   
6562   35000000   326551094   
6563   90000000   289847354   
6564   85000000   194168700   
6565   40000000   109676311   
6566  110000000   574480841   
6567   25000000   135330182   
6568   76000000    69959751   
6569          0           0   
6570  270000000   391081192   
6571   18000000   261572744   
6572   82500000   237681299   
6573          0           0   
6574   52000000   204999686   
6575   45000000   184376254   
6576   80000000   155019340   
6577   50000000   132180323   
6578  100000000   170877916   
6579          0    15309302   
6580   12000000    42931041   
6581   27000000    56308881   
6582   40000000   120455994   
6583   25000000   117719158   
...         ...         ...   
6932          0           0   
6933       2500           0   
6934          0           0   
6935          0           0   
6936          0           0   
6937          0           0   
6938          0           0   
6939          0           0   
6940          0           0   
6941          0           0   
6942          0           0   
6943          0           0   
6944          0           0   
6945      79000      245000   
6946          0           0   
6947          0           0   
6948    1000000        7202   
6949          0           0   
6950          0           0   
6951          0           0   
6952          0           0   
6953          0           0   
6954          0           0   
6955          0           0   
6956          0           0   
6957          0           0   
6958          0           0   
6959          0    13022756   
6960    2700000           0   
6961          0           0   

                                         original_title  runtime  release_year  
6554                              Underworld: Evolution    106.0          2006  
6555         Pirates of the Caribbean: Dead Man's Chest    151.0          2006  
6556                                               Cars    117.0          2006  
6557                                      Casino Royale    144.0          2006  
6558                                  The Da Vinci Code    149.0          2006  
6559                                            Step Up    104.0          2006  
6560                              Ice Age: The Meltdown     91.0          2006  
6561                            Mission: Impossible III    126.0          2006  
6562                              The Devil Wears Prada    109.0          2006  
6563                                       The Departed    151.0          2006  
6564                                        The Holiday    136.0          2006  
6565                                       The Prestige    130.0          2006  
6566                                Night at the Museum    108.0          2006  
6567                                              Babel    143.0          2006  
6568                                    Children of Men    109.0          2006  
6569              American Pie Presents: The Naked Mile     97.0          2006  
6570                                   Superman Returns    154.0          2006  
6571  Borat: Cultural Learnings of America for Make ...     82.0          2006  
6572                                              Click    107.0          2006  
6573                                     Brother Bear 2     75.0          2006  
6574                                       The Break-Up    106.0          2006  
6575                                         Inside Man    129.0          2006  
6576                                     Over the Hedge     83.0          2006  
6577                   Perfume: The Story of a Murderer    147.0          2006  
6578                                      Blood Diamond    14

In [21]:
df5['runtime'].mean()

101.93366093366093

## Ans. Average runtime of the movies in the year 2006 is 101.93366093366093 minutes.

##  Q6. Name any 3 production companies which have invested money in worse revenue movies?


In [22]:
df6=data.drop(columns=['id', 'popularity', 'budget', 'original_title','cast','director', 'keywords', 'runtime', 'genres', 'release_date','vote_count', 'vote_average', 'release_year'])
df6

revenue                               production_companies
0      1513528810  Universal Studios|Amblin Entertainment|Legenda...
1       378436354  Village Roadshow Pictures|Kennedy Miller Produ...
2       295238201  Summit Entertainment|Mandeville Films|Red Wago...
3      2068178225          Lucasfilm|Truenorth Productions|Bad Robot
4      1506249360  Universal Pictures|Original Film|Media Rights ...
5       532950503  Regency Enterprises|Appian Way|CatchPlay|Anony...
6       440603537            Paramount Pictures|Skydance Productions
7       595380321  Twentieth Century Fox Film Corporation|Scott F...
8      1156730962      Universal Pictures|Illumination Entertainment
9       853708609  Walt Disney Pictures|Pixar Animation Studios|W...
10      880674609                       Columbia Pictures|Danjaq|B24
11      183987723  Village Roadshow Pictures|Dune Entertainment|A...
12       36869414  DNA Films|Universal Pictures International (UP...
13      243637091        Columbia Pictures|Happy Madison Productions
14     1405035767  Marvel Studios|Prime Focus|Revolution Sun Studios
15      155760117  Double Feature Films|The Weinstein Company|Fil...
16      325771424  Twentieth Century Fox Film Corporation|M6 Film...
17      518602163                                     Marvel Studios
18      542351353  Walt Disney Pictures|Genre Films|Beagle Pug Fi...
19      650523427  Studio Babelsberg|StudioCanal|Lionsgate|Walt D...
20      209035668                  Walt Disney Pictures|Babieka|A113
21       91709827            Escape Artists|Riche-Ludwig Productions
22      470490832  New Line Cinema|Village Roadshow Pictures|Warn...
23      569651467  Focus Features|Trigger Street Productions|Mich...
24      133346506  Paramount Pictures|Plan B Entertainment|Regenc...
25      682330139  Paramount Pictures|Skydance Productions|China ...
26      215863606  Universal Pictures|Media Rights Capital|Fuzzy ...
27      403802136  Twentieth Century Fox Film Corporation|Marv Fi...
28       88346473  Participant Media|Open Road Films|Anonymous Co...
29      311256926  Gotham Group|Temple Hill Entertainment|TSG Ent...
...           ...                                                ...
10836           0                      Columbia Pictures Corporation
10837           0             Twentieth Century Fox Film Corporation
10838           0                                  Columbia Pictures
10839           0                            Warner Bros. Home Video
10840           0                        Lowndes Productions Limited
10841           0                                      Proteus Films
10842           0                                                NaN
10843           0                  Julian Blaustein Productions Ltd.
10844           0                                   20th Century Fox
10845           0                           Peter Rogers Productions
10846           0     Seven Arts Productions|Hammer Film Productions
10847           0                   Planet Film Productions|Protelco
10848    12000000             Twentieth Century Fox Film Corporation
10849           0                                 Universal Pictures
10850           0                                       Warner Bros.
10851           0                                          High Road
10852           0                              Eden Productions Inc.
10853           0                                                NaN
10854           0     Horizon Pictures|Columbia Pictures Corporation
10855           0                                 Universal Pictures
10856           0                               Walt Disney Pictures
10857           0  Paramount Pictures|Solar Productions|Embassy P...
10858           0                            The Mirisch Corporation
10859           0  Gibraltar Productions|Joel Productions|John Fr...
10860           0  Peter Rogers Productions|Anglo-Amalgamated Fil...
10861           0                                  Bruce Brown Films
10862   

In [27]:
df6.isnull().sum()

revenue                    0
production_companies    1030
dtype: int64

In [35]:
import warnings
warnings.filterwarnings("ignore")

df6.dropna(subset=['production_companies'],inplace=True)
df6

revenue                               production_companies
0      1513528810  Universal Studios|Amblin Entertainment|Legenda...
1       378436354  Village Roadshow Pictures|Kennedy Miller Produ...
2       295238201  Summit Entertainment|Mandeville Films|Red Wago...
3      2068178225          Lucasfilm|Truenorth Productions|Bad Robot
4      1506249360  Universal Pictures|Original Film|Media Rights ...
5       532950503  Regency Enterprises|Appian Way|CatchPlay|Anony...
6       440603537            Paramount Pictures|Skydance Productions
7       595380321  Twentieth Century Fox Film Corporation|Scott F...
8      1156730962      Universal Pictures|Illumination Entertainment
9       853708609  Walt Disney Pictures|Pixar Animation Studios|W...
10      880674609                       Columbia Pictures|Danjaq|B24
11      183987723  Village Roadshow Pictures|Dune Entertainment|A...
12       36869414  DNA Films|Universal Pictures International (UP...
13      243637091        Columbia Pictures|Happy Madison Productions
14     1405035767  Marvel Studios|Prime Focus|Revolution Sun Studios
15      155760117  Double Feature Films|The Weinstein Company|Fil...
16      325771424  Twentieth Century Fox Film Corporation|M6 Film...
17      518602163                                     Marvel Studios
18      542351353  Walt Disney Pictures|Genre Films|Beagle Pug Fi...
19      650523427  Studio Babelsberg|StudioCanal|Lionsgate|Walt D...
20      209035668                  Walt Disney Pictures|Babieka|A113
21       91709827            Escape Artists|Riche-Ludwig Productions
22      470490832  New Line Cinema|Village Roadshow Pictures|Warn...
23      569651467  Focus Features|Trigger Street Productions|Mich...
24      133346506  Paramount Pictures|Plan B Entertainment|Regenc...
25      682330139  Paramount Pictures|Skydance Productions|China ...
26      215863606  Universal Pictures|Media Rights Capital|Fuzzy ...
27      403802136  Twentieth Century Fox Film Corporation|Marv Fi...
28       88346473  Participant Media|Open Road Films|Anonymous Co...
29      311256926  Gotham Group|Temple Hill Entertainment|TSG Ent...
...           ...                                                ...
10728      638641                         Warner Brothers/Seven Arts
10740    14250000                                 Paramount Pictures
10755   181813770  Paramount Pictures|Robert Stigwood Organizatio...
10756   187884007                                 Universal Pictures
10757    55000000                       Laurel Group|Dawn Associates
10758   300218018       Warner Bros.|Dovemead Films|Film Export A.G.
10759    70000000  Compass International Pictures|Falcon Internat...
10760   141000000  Universal Pictures|Oregon Film Factory|Stage I...
10761    49579269  United Artists|Jewel Productions|Pimlico Films...
10762    50000000                  Universal Pictures|EMI Films Ltd.
10764    41590893                                 Paramount Pictures
10766     7185518                                          Image Ten
10770    35000000             Columbia Pictures|Casablanca Filmworks
10771    30471420                                      Fantasy Films
10775    14560084                 EMI Films Ltd.|Mersham Productions
10777          11                                chateau productions
10778     7230000                      Columbia Pictures Corporation
10779    22765081                                     EMI Films Ltd.
10780    24046533                            United Artists|Solofilm
10788    21049053              Universal Pictures|Motown Productions
10791    26518355  Twentieth Century Fox Film Corporation|Mace Ne...
10793     3713768         Nepenthe Productions|Watership Productions
10812     5438927         Universal Pictures|The Mirisch Corporation
10817      321952                     FM Productions|Last Waltz Inc.
10819    78000000                                       Warner Bros.
10822    33736689                               Chenault Productions
10828   

In [25]:
df6=df6[(df6['revenue']>0)]
df6

revenue                               production_companies
0      1513528810  Universal Studios|Amblin Entertainment|Legenda...
1       378436354  Village Roadshow Pictures|Kennedy Miller Produ...
2       295238201  Summit Entertainment|Mandeville Films|Red Wago...
3      2068178225          Lucasfilm|Truenorth Productions|Bad Robot
4      1506249360  Universal Pictures|Original Film|Media Rights ...
5       532950503  Regency Enterprises|Appian Way|CatchPlay|Anony...
6       440603537            Paramount Pictures|Skydance Productions
7       595380321  Twentieth Century Fox Film Corporation|Scott F...
8      1156730962      Universal Pictures|Illumination Entertainment
9       853708609  Walt Disney Pictures|Pixar Animation Studios|W...
10      880674609                       Columbia Pictures|Danjaq|B24
11      183987723  Village Roadshow Pictures|Dune Entertainment|A...
12       36869414  DNA Films|Universal Pictures International (UP...
13      243637091        Columbia Pictures|Happy Madison Productions
14     1405035767  Marvel Studios|Prime Focus|Revolution Sun Studios
15      155760117  Double Feature Films|The Weinstein Company|Fil...
16      325771424  Twentieth Century Fox Film Corporation|M6 Film...
17      518602163                                     Marvel Studios
18      542351353  Walt Disney Pictures|Genre Films|Beagle Pug Fi...
19      650523427  Studio Babelsberg|StudioCanal|Lionsgate|Walt D...
20      209035668                  Walt Disney Pictures|Babieka|A113
21       91709827            Escape Artists|Riche-Ludwig Productions
22      470490832  New Line Cinema|Village Roadshow Pictures|Warn...
23      569651467  Focus Features|Trigger Street Productions|Mich...
24      133346506  Paramount Pictures|Plan B Entertainment|Regenc...
25      682330139  Paramount Pictures|Skydance Productions|China ...
26      215863606  Universal Pictures|Media Rights Capital|Fuzzy ...
27      403802136  Twentieth Century Fox Film Corporation|Marv Fi...
28       88346473  Participant Media|Open Road Films|Anonymous Co...
29      311256926  Gotham Group|Temple Hill Entertainment|TSG Ent...
...           ...                                                ...
10740    14250000                                 Paramount Pictures
10755   181813770  Paramount Pictures|Robert Stigwood Organizatio...
10756   187884007                                 Universal Pictures
10757    55000000                       Laurel Group|Dawn Associates
10758   300218018       Warner Bros.|Dovemead Films|Film Export A.G.
10759    70000000  Compass International Pictures|Falcon Internat...
10760   141000000  Universal Pictures|Oregon Film Factory|Stage I...
10761    49579269  United Artists|Jewel Productions|Pimlico Films...
10762    50000000                  Universal Pictures|EMI Films Ltd.
10764    41590893                                 Paramount Pictures
10766     7185518                                          Image Ten
10770    35000000             Columbia Pictures|Casablanca Filmworks
10771    30471420                                      Fantasy Films
10775    14560084                 EMI Films Ltd.|Mersham Productions
10777          11                                chateau productions
10778     7230000                      Columbia Pictures Corporation
10779    22765081                                     EMI Films Ltd.
10780    24046533                            United Artists|Solofilm
10787    16393000                                                NaN
10788    21049053              Universal Pictures|Motown Productions
10791    26518355  Twentieth Century Fox Film Corporation|Mace Ne...
10793     3713768         Nepenthe Productions|Watership Productions
10812     5438927         Universal Pictures|The Mirisch Corporation
10817      321952                     FM Productions|Last Waltz Inc.
10819    78000000                                       Warner Bros.
10822    33736689                               Chenault Productions
10828   

In [26]:
df6.nunique()

revenue                 4701
production_companies    3682
dtype: int64

In [28]:
#We are calculating the mean revenue of each production company. As there are 3682 different production companies, we should get 3682 rows.

df6_=df6.groupby('production_companies').mean()
df6_

revenue
production_companies                                            
10th Hole Productions|Benaroya Pictures|Four of...  1.238200e+05
120dB Films|Scion Films|Voodoo Production Servi...  1.378000e+03
1492 Pictures|Dune Entertainment|Fox 2000 Pictu...  1.745788e+08
1492 Pictures|Warner Bros.|Heyday Films             9.764756e+08
1492 Pictures|Warner Bros.|Heyday Films|MIRACLE...  8.766885e+08
1492 Pictures|Warner Bros.|Heyday Films|P of A ...  7.898046e+08
1818|Lone Wolf McQuade Associates|Topkick Produ...  1.223263e+07
20th Century Fox                                    7.108200e+07
20th Century Fox|1492 Pictures                      5.409517e+06
20th Century Fox|American Entertainment Partner...  2.520546e+07
20th Century Fox|Baltimore Pictures                 2.327893e+07
20th Century Fox|Chernin Entertainment|Walden M...  1.197722e+08
20th Century Fox|Davis Entertainment                2.243328e+07
20th Century Fox|Double Feature Films|Appian Wa...  6.261665e+07
20th Century Fox|Egg Pictures                       1.066838e+08
20th Century Fox|Figment Films                      1.440569e+08
20th Century Fox|Gladden Entertainment              4.272120e+07
20th Century Fox|Gran Via Productions|Walden Media  5.310554e+06
20th Century Fox|Gruskoff/Venture Films             2.269769e+07
20th Century Fox|Largo Entertainment|JVC Entert...  5.713872e+07
20th Century Fox|Morgan Creek Productions           2.938196e+07
20th Century Fox|Penta Pictures                     7.549350e+05
20th Century Fox|Regency Enterprises                4.925430e+07
20th Century Fox|Regency Enterprises|Guy Walks ...  5.065008e+07
20th Century Fox|Regency Enterprises|New Regenc...  4.398989e+06
20th Century Fox|Saban Entertainment                1.797974e+07
20th Century Fox|Sandollar Productions              2.162100e+07
20th Century Fox|Turner Pictures                    1.367069e+07
20th Century Fox|Village Roadshow Pictures|NPV ...  1.000201e+08
20th Century Fox|Weed Road Pictures|Dutch Oven|...  4.782075e+08
...                                                          ...
Working Title Films|Translux                        8.710045e+07
Worldview Entertainment                             6.427467e+04
Worldview Entertainment|Arcade Pictures             9.221000e+03
Worldview Entertainment|Dragonfly Entertainment...  6.833445e+06
Worldview Entertainment|Muskat Filmed Propertie...  2.365467e+06
Worldview Entertainment|New Regency Pictures|TS...  1.032151e+08
XYZ Films                                           4.997000e+04
Yari Film Group                                     4.534909e+06
Yari Film Group|Furst Films                         3.822670e+05
Yash Raj Films                                      4.639643e+07
Zazen Produções                                   2.172010e+05
Zenith Entertainment|Initial Pictures|U.K. Prod...  2.826523e+06
Zentropa Entertainments                             2.227167e+06
Zentropa Entertainments|BBC Films|Scottish Scre...  9.080000e+05
Zentropa Entertainments|Danmarks Radio (DR)|Art...  6.749180e+05
Zentropa Entertainments|Det Danske Filminstitut...  7.918670e+05
Zentropa Entertainments|Les Films du Losange|Co...  1.028590e+07
Zero Gravity Management|Farraj Factory|Kinetic ...  2.710000e+03
Zininsa Film Production|Intellectual Properties...  4.040588e+06
Zoetrope Studios                                    8.347151e+07
Zoetrope Studios|Golan-Globus Productions           3.221568e+06
Zucker Brothers Productions                         7.027290e+06
Zweites Deutsches Fernsehen (ZDF)|Les Films Ari...  7.720000e+07
abramoff Production|Scorpion Film Production|Ab...  4.192440e+06
area japan                                          3.170000e+05
chateau productions                                 1.100000e+01
creative differences|History films                  6.467348e+06
erbp                                                4.440980e+05
iDeal Partners Film Fund                            5.512500e+04
icon                       

In [33]:
df6_=df6_.sort_values('revenue')
df6_.head(10)

revenue
production_companies                                       
Gramercy Pictures|Alphaville Films|View Askew P...      2.0
Lions Gate Films|Cruise/Wagner Productions|Baum...      2.0
Tales From The Crypt Holdings|Universal City St...      5.0
Studio 4°C                                             5.0
Det Danske Filminstitut|Spring Creek Production...      6.0
Filmhuset Gruppen|Leiknar Myndir                       11.0
chateau productions                                    11.0
Alcon Entertainment|ETIC Films|C.R.G. Internati...     12.0
Paramount Pictures|Universal Pictures                  13.0
New Line Cinema|Sean S. Cunningham Films               15.0

## Ans. The production companies with worst revenues are:
##  1.Gramercy Pictures|Alphaville Films|View Askew Productions
##  2.Lions Gate Films|Cruise/Wagner Productions|Baumgarten Merims Productions
##  3.Tales From The Crypt Holdings|Universal City Studios
